electricitylci/main.py

```diff
@@ -67,6 +67,20 @@ def main():
     generation_mix_dict = electricitylci.write_process_dicts_to_jsonld(
         generation_mix_dict
     )
+    print("us average mix to dict")
```

Comment:
I think this should all get moved down to line 117 (before `write_surplus_pool_and_consumption_mix_dict()`), so that this doesn't run for EPA trading is false. Running ELCI_1 now to confirm.

Reply:
This is for sure having impacts on ELCI_1: 
```
WARNING:generation_mix.py:olcaschema_usaverage:Trouble matching dictionary for creating fuel mix COAL - AZNM.Skipping this flow for now
...
WARNING:generation_mix.py:olcaschema_usaverage:Trouble matching dictionary for creating fuel mix OTHF - SRVC.Skipping this flow for now
```
The US average profiles are being created, but not used in the mixes (as expected). The original "ALL" profiles are still being created and used, so functionally ELCI_1 looks okay - the console output is just messy.

Reply:
I think TJ created this warning to indicate where some blanks were occurring. @TJTapajyoti , please confirm.

The US average generation profiles should be used as providers for the Canadian and Mexico regions.

Reply:
@jump2conclusionsmatt  Hi Matt, I realize the problem here. Basically if you look at the _init_.py file, you will see that I have created two new functions. 

`def write_fuel_mix_database_to_dict(
    genmix_database, gen_dict, regions=None
):`

`def write_international_mix_database_to_dict(
    genmix_database, usfuelmix_dict, regions=None
):`

These two new functions work specifically with ELCI 3 and is not compatible with ELCI 1 or 2 because the methodology is completely different of how the generationmix.py. Two new functions pertaining only to ELCI3 have been introduced into the generation mix.py - 

`from electricitylci.generation_mix import olcaschema_international;`
`from electricitylci.generation_mix import olcaschema_usaverage`

Thus I think the best way to solve the current problem is to remove these lines from the __init__.py file from both the functions of fuel_mix and international_mix. If these functions are called for ELCI1 (regions other than eGRID Subregions), we should just skip the calculations. 

`    if regions is None:
        regions = config.model_specs.regional_aggregation
    if regions in ["FERC","US","BA"]:
        usaverage_dict = olcaschema_usaverage(
                genmix_database, gen_dict, subregion="BA"
        )
    else:`


change to - 
`    if regions is egrid_subregions:
        usaverage_dict = olcaschema_usaverage(
                genmix_database, gen_dict, subregion=regions
        return usaverage_dict 
        )

    else:
        return None`

However, the challenge comes if returning None will cause problems in these lines in the main.py  - 
`    usavegfuel_mix_dict = electricitylci.write_process_dicts_to_jsonld(
        usavegfuel_mix_dict
    )  `

`    international_mix_dict = electricitylci.write_process_dicts_to_jsonld(
    international_mix_dict`

How do you recommend we address this @jump2conclusionsmatt ?

Reply:
Fixing the problem by moving the calls for us average mix dictionary creation and international mix creating withing the else: section of the main.py file which runs only when the epa_egrid_trading is False.
Not removing the lines from the __init__.py file as of now.